src/helpers/issue.ts

```diff
@@ -683,23 +683,38 @@ export const getOpenedPullRequestsForAnIssue = async (issueNumber: number, userN
   });
 };
 
-export const getOpenedPullRequests = async (username: string) => {
+export const getOpenedPullRequests = async (username: string, state: PullRequestState) => {
   const context = getBotContext();
   const prs = await getAllPullRequests(context, "open");
-  return prs.filter((pr) => !pr.draft && (pr.user?.login === username || !username));
+  return prs.filter(
+    (pr) => (state === PullRequestState.READY ? !pr.draft : state === PullRequestState.DRAFT ? pr.draft : true) && (pr.user?.login === username || !username)
+  );
 };
 
 export const getCommitsOnPullRequest = async (pullNumber: number) => {
   const logger = getLogger();
   const context = getBotContext();
   const payload = getBotContext().payload as Payload;
   try {
-    const { data: commits } = await context.octokit.rest.pulls.listCommits({
-      owner: payload.repository.owner.login,
-      repo: payload.repository.name,
-      pull_number: pullNumber,
-    });
-    return commits;
+    const perPage = 100;
+    let curPage = 1;
+    const allCommits = [];
+    let fetchDone = false;
+    while (!fetchDone) {
+      const response = await context.octokit.rest.pulls.listCommits({
+        owner: payload.repository.owner.login,
+        repo: payload.repository.name,
+        pull_number: pullNumber,
+        per_page: 100,
+        page: curPage,
+      });
+      allCommits.push(...response.data);
+      if (response.data.length < perPage) {
+        fetchDone = true;
+        return allCommits;
```

Comment:
If we want to return allCommits when the `fetchDone` is true, no need to define the local variable `fetchDone`, can do without it.

Reply:
https://github.com/ubiquity/ubiquibot/pull/723/commits/bc6406cade78c3212f5ddfe3b601391ff856d67a

fixed here